include/benchmark/benchmark_api.h

```diff
@@ -230,7 +230,7 @@ BENCHMARK_UNUSED static int stream_init_anchor = InitializeStreams();
 // expression from being optimized away by the compiler. This function is
 // intended to add little to no overhead.
 // See: https://youtu.be/nXaxk27zwlk?t=2441
-#if defined(__GNUC__)
+#if defined(__GNUC__) && !defined(__pnacl__) && !defined(EMSCRIPTEN)
```

Comment:
In case of PNaCl it fails to link, with an explicit message blaming inline assembly:
```
[21/29] LINK build/init-bench.bc
FAILED: /Users/marat/Projects/fxdiv-confu/build/init-bench.bc
/Users/marat/Tools/nacl_sdk/pepper_49/toolchain/mac_pnacl/bin/pnacl-clang++   -o /Users/marat/Projects/fxdiv-confu/build/init-bench.bc /Users/marat/Projects/fxdiv-confu/build/bench/init.cc.bc /Users/marat/Projects/fxdiv-confu/lib/libgooglebenchmark.a
Function _ZL13init_uint32_tRN9benchmark5StateE disallowed: bad result type: %struct.fxdiv_divisor_uint32_t*   %divisor.bc2 = bitcast i8* %divisor to %struct.fxdiv_divisor_uint32_t*
Function _ZL13init_uint32_tRN9benchmark5StateE disallowed: inline assembly:    call void asm sideeffect "", "*imr,~{memory}"(%struct.fxdiv_divisor_uint32_t* %divisor.bc2), !dbg !100433
```

In case of Emscripten/Asm.js, it links, but fails with a strange error at runtime:
```
Run on (1 X 1000.15 MHz CPU )
2017-02-10 21:33:11
Benchmark                           Time           CPU Iterations
-----------------------------------------------------------------
exception thrown: ReferenceError: $5 is not defined,ReferenceError: $5 is not defined
    at Array.Rb (/Users/marat/Projects/fxdiv-confu/bin/multiply-bench.js:8:434)
    at Array.Pc (/Users/marat/Projects/fxdiv-confu/bin/multiply-bench.js:8:36040)
    at mc (/Users/marat/Projects/fxdiv-confu/bin/multiply-bench.js:8:17877)
    at ic (/Users/marat/Projects/fxdiv-confu/bin/multiply-bench.js:8:10462)
    at hc (/Users/marat/Projects/fxdiv-confu/bin/multiply-bench.js:8:5620)
    at Object.Wb [as _main] (/Users/marat/Projects/fxdiv-confu/bin/multiply-bench.js:8:1543)
    at Object.callMain (/Users/marat/Projects/fxdiv-confu/bin/multiply-bench.js:18:5036)
    at doRun (/Users/marat/Projects/fxdiv-confu/bin/multiply-bench.js:18:5742)
    at run (/Users/marat/Projects/fxdiv-confu/bin/multiply-bench.js:18:5910)
    at Object.<anonymous> (/Users/marat/Projects/fxdiv-confu/bin/multiply-bench.js:18:7050)
/Users/marat/Projects/fxdiv-confu/bin/multiply-bench.js:1
(function (exports, require, module, __filename, __dirname) { var Module;if(!Module)Module=(typeof Module!=="undefined"?Module:null)||{};var moduleOverrides={};for(var key in Module){if(Module.hasOwnProperty(key)){moduleOverrides[key]=Module[key]}}var ENVIRONMENT_IS_WEB=false;var ENVIRONMENT_IS_WORKER=false;var ENVIRONMENT_IS_NODE=false;var ENVIRONMENT_IS_SHELL=false;if(Module["ENVIRONMENT"]){if(Module["ENVIRONMENT"]==="WEB"){ENVIRONMENT_IS_WEB=true}else if(Module["ENVIRONMENT"]==="WORKER"){ENVIRONMENT_IS_WORKER=true}else if(Module["ENVIRONMENT"]==="NODE"){ENVIRONMENT_IS_NODE=true}else if(Module["ENVIRONMENT"]==="SHELL"){ENVIRONMENT_IS_SHELL=true}else{throw new Error("The provided Module['ENVIRONMENT'] value is not valid. It must be one of: WEB|WORKER|NODE|SHELL.")}}else{ENVIRONMENT_IS_WEB=typeof window==="object";ENVIRONMENT_IS_WORKER=typeof importScripts==="function";ENVIRONMENT_IS_NODE=typeof process==="object"&&typeof require==="function"&&!ENVIR

ReferenceError: $5 is not defined
    at Array.Rb (/Users/marat/Projects/fxdiv-confu/bin/multiply-bench.js:8:434)
    at Array.Pc (/Users/marat/Projects/fxdiv-confu/bin/multiply-bench.js:8:36040)
    at mc (/Users/marat/Projects/fxdiv-confu/bin/multiply-bench.js:8:17877)
    at ic (/Users/marat/Projects/fxdiv-confu/bin/multiply-bench.js:8:10462)
    at hc (/Users/marat/Projects/fxdiv-confu/bin/multiply-bench.js:8:5620)
    at Object.Wb [as _main] (/Users/marat/Projects/fxdiv-confu/bin/multiply-bench.js:8:1543)
    at Object.callMain (/Users/marat/Projects/fxdiv-confu/bin/multiply-bench.js:18:5036)
    at doRun (/Users/marat/Projects/fxdiv-confu/bin/multiply-bench.js:18:5742)
    at run (/Users/marat/Projects/fxdiv-confu/bin/multiply-bench.js:18:5910)
    at Object.<anonymous> (/Users/marat/Projects/fxdiv-confu/bin/multiply-bench.js:18:7050)
```

Both Native Client SDK and Emscripten are based on forks of Clang, not upstream Clang. The lack of inline assembly support is documented ([for PNaCl](https://developer.chrome.com/native-client/reference/pnacl-c-cpp-language-support#inline-assembly), [for Emscripten](https://kripken.github.io/emscripten-site/docs/getting_started/FAQ.html#why-does-my-code-fail-to-compile-with-an-error-message-about-inline-assembly-or-text-asm)) so I don't think they will fix it.